app/controllers/sign_up/completions_controller.rb

```diff
@@ -3,7 +3,8 @@ class CompletionsController < ApplicationController
     include SecureHeadersConcern
 
     before_action :confirm_two_factor_authenticated
-    before_action :verify_confirmed, if: :ial2?
+    before_action :confirm_identity_verified, if: :ial2?
```

Comment:
I am not sure the completions controller is the correct place for this logic. The completions controller is not what is responsible for redirecting the user to the service provider or checking that the user has satisfied all of the requirements for being redirected. That is the `OpenidConnect::AuthorizationController#index` and the `SamlIdpController#auth` action.

Reply:
The before_action on line 6 was already there. I just renamed it. Do you recommend removing that, or leaving it alone, or adding the before_action here and also modifying the places you mentioned?

Reply:
I looked at the latest changes and I think it is probably fine to leave those checks in here. They are shortcutting the effect that an eventual redirect to the authorization controller or SAML IdP controller would have.